tools/cli/workflowCommands.go

```diff
@@ -1904,60 +1904,41 @@ func getResetEventIDByType(
 		if err != nil {
 			return
 		}
+	case resetTypeFirstDecisionScheduled:
+		resetBaseRunID, decisionFinishID, err = getFirstDecisionTaskByType(ctx, domain, wid, rid, frontendClient, types.EventTypeDecisionTaskScheduled)
+		if err != nil {
+			return
+		}
+		// decisionFinishID is exclusive in reset API
+		decisionFinishID++
+	case resetTypeLastDecisionScheduled:
+		resetBaseRunID, decisionFinishID, err = getLastDecisionTaskByType(ctx, domain, wid, rid, frontendClient, types.EventTypeDecisionTaskScheduled)
+		if err != nil {
+			return
+		}
+		// decisionFinishID is exclusive in reset API
+		decisionFinishID++
 	default:
 		panic("not supported resetType")
 	}
 	return
 }
 
-func getEarliestDecisionID(
+func getFirstDecisionTaskByType(
 	ctx context.Context,
-	domain string, wid string,
-	rid string, earliestTime int64,
+	domain string,
+	workflowID string,
+	runID string,
 	frontendClient frontend.Client,
-) (decisionFinishID int64, err error) {
-	req := &types.GetWorkflowExecutionHistoryRequest{
-		Domain: common.StringPtr(domain),
-		Execution: &types.WorkflowExecution{
-			WorkflowID: common.StringPtr(wid),
-			RunID:      common.StringPtr(rid),
-		},
-		MaximumPageSize: common.Int32Ptr(1000),
-		NextPageToken:   nil,
-	}
-
-OuterLoop:
-	for {
-		resp, err := frontendClient.GetWorkflowExecutionHistory(ctx, req)
-		if err != nil {
-			return 0, printErrorAndReturn("GetWorkflowExecutionHistory failed", err)
-		}
-		for _, e := range resp.GetHistory().GetEvents() {
-			if e.GetEventType() == types.EventTypeDecisionTaskCompleted {
-				if e.GetTimestamp() >= earliestTime {
-					decisionFinishID = e.GetEventID()
-					break OuterLoop
-				}
-			}
-		}
-		if len(resp.NextPageToken) != 0 {
-			req.NextPageToken = resp.NextPageToken
-		} else {
-			break
-		}
-	}
-	if decisionFinishID == 0 {
-		return 0, printErrorAndReturn("Get DecisionFinishID failed", fmt.Errorf("no DecisionFinishID"))
-	}
-	return
-}
+	decisionType types.EventType,
+) (resetBaseRunID string, decisionFinishID int64, err error) {
 
-func getLastDecisionCompletedID(ctx context.Context, domain, wid, rid string, frontendClient frontend.Client) (decisionFinishID int64, err error) {
+	resetBaseRunID = runID
```

Comment:
I think this return value was remove in a previous diff as it's value is always equal to the input runID. Any reason for adding it back?

Reply:
You are correct. It reverted during conflict resolution. I updated it